ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform
+package db.migration.translation
+
+import java.io.InputStream
+
+import com.daml.lf.archive.{Decode, Reader}
+import com.daml.lf.value.Value.{ContractId, VersionedValue}
+import com.daml.lf.value.{ValueCoder, ValueOuterClass}
+import org.slf4j.LoggerFactory
+
+private[migration] object ValueSerializer {
+
+  private[this] val logger = LoggerFactory.getLogger(getClass)
+
+  private[translation] object DeprecatedValueVersionsError {
+    val DeprecatedValueVersions = Set("1", "2", "3", "4", "5")
+    val UnsupportedErrorMessage = """Unsupported value version (\d)""".r
+
+    def unapply[X](arg: Either[ValueCoder.DecodeError, X]): Option[String] =
+      arg match {
+        case Left(ValueCoder.DecodeError(UnsupportedErrorMessage(version)))
+            if DeprecatedValueVersions(version) =>
+          Some(version)
+        case _ => None
+      }
+  }
+
+  private[translation] def handleDeprecatedValueVersions[X](
+      x: Either[ValueCoder.DecodeError, X],
+  ): Either[ValueCoder.DecodeError, X] = {
+    x match {
+      case DeprecatedValueVersionsError(deprecateVersion) =>
+        logger.error(
+          s"*** Deserialization of value version $deprecateVersion is not supported by the SDK 1.9.0 or latter. ***")
+        logger.error(
+          s"*** Please upgrade your sandbox database by upgrading your SDK to 1.7 or 1.8 first. ***")
```

Comment:
@gerolf-da Would it be better to throw a error or add a System.exit(1) here

Reply:
Never ever use `System.exit` in a library. You don't know where the library is used. I'd rather not kill an unknown process.

Reply:
:+1: